nltk/data.py

```diff
@@ -73,7 +73,7 @@
 
 # User-specified locations:
 path += [d for d in os.environ.get('NLTK_DATA', str('')).split(os.pathsep) if d]
-if os.path.expanduser('~/') != '~/':
+if not os.environ.get('APPENGINE_RUNTIME') and os.path.expanduser('~/') != '~/':
```

Comment:
maybe ``'APPENGINE_RUNTIME' in os.environ`` is a clearer version.

However, if it's set to an empty string, behavior will be different.